server/src/api/read-schematic-collections.ts

```diff
@@ -22,10 +27,30 @@ interface Schematic {
   }[];
 }
 
-export function readSchematicCollections(
+export function readAllSchematicCollections(basedir: string) {
+  try {
+    const nodeModulesDir = path.join(basedir, 'node_modules');
+    const packages = listOfUnnestedNpmPackages(nodeModulesDir);
+    const schematicCollections = packages.filter(p => {
+      try {
+        return !!readJsonFile(path.join(p, 'package.json'), nodeModulesDir).json
+          .schematics;
+      } catch (e) {
```

Comment:
Can you leave a comment saying what type of error we expect here? If the type of errors are unexpected maybe we should log them rather than silently filtering them out

Reply:
Added a check to ignore only the right type of error